### DISCUSSION TITLE
SSG rendering a Gollum Wiki

### DISCUSSION THREAD

**Discussion by jgsuess** [gollum/gollum#1816] [Q&A]
I have a Gitlab wiki, which renders nicely in Gollum and would like to make it static like an SSG does. Is there any mode that allows Gollum to be used in this fashion?

**Comment by dometto** [comment]
Hi @jgsuess! Static site generation is not currently a feature of gollum. There used to be some projects that provided this (`gollum-site` and `jekyll-gollum`) but they seem to be deprecated. I wonder how difficult it would be to write a `jekyll` plugin for gollum now that gollum also supports YAML metadata. Basically, it would require `jekyll` transforming the content of each file in the site by passing it through gollum's [filter chain](https://github.com/gollum/gollum-lib/blob/master/lib/gollum-lib/markup.rb#L122) (taking account of the configuration of the wiki). If anyone is interested in working on this I would be happy to assist!

You can of course already use `jekyll` on your gollum wiki's git directory and maybe get a decent result, but `jekyll` would just not render any gollum-specific markup (e.g. UML, gollum-specific tags).
